evm/vm/message.py

```diff
@@ -187,12 +196,34 @@ def __init__(self,
             validate_access_list(access_list)
         self.access_list = access_list
 
+    @property
+    def paygas_total_fee(self):
+        return self.paygas_gasprice * self.transaction_gas_limit
+
+    @property
+    def paygas_gasprice(self):
+        if self._paygas_gasprice is None:
+            return 0
```

Comment:
I'm still curious about returning `0` instead of `None` here.  It seems valuable to be able to distinguish between *not-set* and *set-to-0* which cannot be done in the current implementation.

Reply:
My rationale was that from inside(`Message` object) the two cases are distinguished by `msg._paygas_gasprice` and from outside one indeed can not tell the difference. But since when `msg.paygas_gasprice` is used to compute gas charge/refund (from outside) the two cases are basically the same.

Also if `msg.paygas_gasprice` could return `None`, we need to add a condition check every time it's used to compute gas charge/refund.
```python
gas_refund_amount = (gas_refund + gas_remaining) * computation.msg.paygas_gasprice if computation.msg.paygas_gasprice is not None else 0
```

Reply:
So i'm on the fence about this.  Your call.  Here's some extra thoughts on why it *might* still be good to use `None`.

The `Computation` object has an existing convention of having `get_XXX` methods for things to account for these little bits of internal logic.  I think the following API is consisten with the rest of the class and clean.

```python
class Computation:
    ...
    def get_PAYGAS_gas_price(self):
        if self._paygas_gasprice is None:
            return 0
        else:
            return self._paygas_gasprice

# and then used elsewhere
gas_refund_amount = (gas_refund + gas_remaining) * computation.get_paygas_gasprice()
```

It also has the nice benefit of not reaching down into the message object.

Again, your call.  I'm +0 on this change if you don't think there is a valid use case for differentiating between set to 0 and unset.